### ISSUE TITLE
stacktrace when using from the command line

### ISSUE THREAD

**Issue by ghost** [loomchild/segment#15]
It is relatively rare, but is irritating when segmenting a large corpus:

java.lang.NullPointerException
        at net.loomchild.segment.srx.SrxTextIterator.next(SrxTextIterator.java:216)
        at net.loomchild.segment.ui.console.Segment.performSegment(Segment.java:604)
        at net.loomchild.segment.ui.console.Segment.createAndSegment(Segment.java:493)
        at net.loomchild.segment.ui.console.Segment.segment(Segment.java:193)
        at net.loomchild.segment.ui.console.Segment.run(Segment.java:115)
        at net.loomchild.segment.ui.console.Segment.main(Segment.java:88)

These are the 'lines of text' that cause a crash: 
[problems.zip](https://github.com/loomchild/segment/files/2697704/problems.zip)

**Comment by loomchild** [comment]
Thank you very much for bug report, I will look into this ASAP.

**Comment by ghost** [comment]
The issue is not in the .srx; it does this with the built-in srx as well. It does so with both Dutch and English.

**Comment by loomchild** [comment]
Hm, I can't reproduce the issue. 

Could you share how do you exactly get the issue? When I run below script for each file you have attached, I don't have any errors:

`bin/segment --buffer-length=1000000 -i <file-name>`

What's weird is that I don't really see how I can get an NullPointerException in line 216 of SrxTextIterator.java:

`start = 0;`

That suggests that you are using old or modified version of Segment. Could you please share the version you are using or the source code?

**Comment by ghost** [comment]
I did not add the buffer=length parameter. Why should I? If it detects buffer is too short, it could lengthen it, couldn't it? (I am just a user...) Or set it to a % of available memory.
But anyway... adding this --buffer-length does the trick.

**Comment by loomchild** [comment]
I added it since I had 'Buffer too short' error. The buffer is not dynamic, it's fixed. By default it's just 64K and it needs to be as long as the longest segment. In the texts that you shared there was no separator, so the buffer was not large enough.

In latest unreleased version I improved the error message to make it more clear, since another user was a bit confused. I will increase the default buffer size and make a release.
